### DISCUSSION TITLE
Current behaviour of table with selection​ may confuse the users

### DISCUSSION THREAD

**Discussion by kajanan1212** [h2oai/wave#1852] [Ideas]
### Actual behaviours:

**1. We can select/unselect any number of rows by clicking on the checkbox without any confusion.**

https://user-images.githubusercontent.com/89178049/221090525-44d8072e-7885-40ac-82e8-ce032ac7fda2.mov

<br>

**2. Clicking on the row to select it unselects the already selected rows.**

https://user-images.githubusercontent.com/89178049/221088240-56c9c1f2-5d6c-4c12-acda-225e61cbcbf0.mov

<br>

**3. Clicking on the row to unselect it unselects all the selected rows excluding clicked one.**

https://user-images.githubusercontent.com/89178049/221090532-585006a1-23f6-407b-ac15-bb347c02251d.mov

<br>

### In my opinion, the expected behaviour should be either

1. We should be able to select/unselect any number of rows by clicking on the checkbox or row.

2. Or we shouldn’t be able to select/unselect any by clicking on the row. Clicking on the checkbox should be compulsory to select/unselect.

### Steps To Reproduce

```py
from h2o_wave import main, app, Q, ui


@app('/')
async def serve(q: Q):
    q.page['example'] = ui.form_card(box='1 1 3 4', items=[
    ui.table(name='table', multiple=True, checkbox_visibility='always', columns=[
        ui.table_column(name='name', label='Name', link=False),
        ui.table_column(name='surname', label='Surname'),
    ], rows=[
        ui.table_row(name='row1', cells=['John', 'Doe']),
        ui.table_row(name='row2', cells=['Alice', 'Smith']),
        ui.table_row(name='row3', cells=['Bob', 'Adams']),
    ])
])

    await q.page.save()
```

**Comment by mturoci** [comment]
That's how selection works in general (both Windows / Mac):


https://user-images.githubusercontent.com/64769322/221119836-71190458-00a0-410f-8ec8-c9bae1040795.mov

**Reply by vopani** [reply]
Don't think this is a fair comparison due to the absence of checkbox.

I agree with @kajanan1212 , IMO the behaviour should change as he suggests.
eg: In most apps that allow multiselect, it works as he's suggested.

**Reply by dulajra** [reply]
Agree with @vopani and @kajanan1212 

The behaviour should be as @kajanan1212 explained when the checkboxes are enabled. A good example is the [Google Files](https://play.google.com/store/apps/details?id=com.google.android.apps.nbu.files&hl=en&gl=US) app

In the Mac Finder app, the checkbox-enabled mode is simulated when the Cmd key is pressed. When the Cmd key is pressed if you click anywhere on the row the row will be selected/deselected.
